www/README.md

```diff
@@ -0,0 +1,121 @@
+
+```
+npm start
+```
+
+The `gulp` command will start a file watcher which will update the generated `app` code after any changes are saved. Note: gulp file watcher does not currently support adding or deleting files, this will require a restart of gulp). Two new directories will also be created in the project.
+
+* `master/node_modules` - contains npm dependencies
+* `master/bower_components` - contains the angular framework files and any custom dependencies
+
+Bower components should be refernced in one of the `vendor.json` files below:
+
+* `master/vendor.base.json` - 3rd party vendor javascript required to start the app. JS is compiled to `base.js` and loaded before `app.js`
+* `master/vendor.json` - 3rd party vendor scripts to make the app work, usually lazy loaded. Can be js or css. Copied to `vendor/*`.
+
+### Serving the app during development
+
+The app can be served through `kubectl`, but for some types of review a local web server is convenient. One can be installed as follows:
+
+```
+sudo npm install -g http-server
+```
+
+The server can then be launched:
+
+```
+cd app
+http-server -a localhost -p 8000
+```
+
+### Configuration
+#### Configuration settings
+A json file can be used by `gulp` to automatically create angular constants. This is useful for setting per environment variables such as api endpoints.
+*  ```www/master/shared/config/development.json``` or ```www/master/shared/config/production.json``` can be created from the ```www/master/shared/config/development.example.json``` file.
+* ```development.example.json``` should be kept up to date with default values, since ```development.json``` is not under source control.
+* Component configuration can be added to ```www/master/components/<component name>/config/development.json``` and it will be combined with the main app config files and compiled into the intermediary ```www/master/shared/config/generated-config.js``` file.
+* All ```generated-config.js``` is compiled into ```app.js```
+* Production config can be generated using ```gulp config --env production``` or ```gulp --env production```
+* The generated angular constant is named ```ENV``` with the shared root and each component having their own child configuration. For example,
+```
+www/master
+├── shared/config/development.json
+└── components
+    ├── dashboard/config/development.json
+    ├── graph/config/development.json
+    └── my_component/config/development.json
+```
+produces ```www/master/shared/config/generated-config.js```:
+```
+angular.module('kubernetesApp.config', [])
+.constant('ENV', {
+  '/': <www/master/shared/config/development.json>,
+  'dashboard': <www/master/components/dashboard/config/development.json>,
+  'graph': <www/master/components/graph/config/development.json>,
+  'my_component': <www/master/components/my_component/config/development.json>
+});
+```
+
+#### Kubernetes server configuration
+
+**RECOMMENDED**: By default the Kubernetes api server does not support CORS,
```

Comment:
This is correct but should be an irrelevant detail, because apiserver should be hosting the UI itself?